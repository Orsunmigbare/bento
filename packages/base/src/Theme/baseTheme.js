import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
