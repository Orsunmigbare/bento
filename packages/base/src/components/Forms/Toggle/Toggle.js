import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const Toggle = ({
  children,
  id,
  group,
  disabled,
  onChange,
  className,
  defaultChecked,
  ...otherProps
}) => {
  return (
    <CheckWrapper htmlFor={id} className={className}>
      <Check
        id={id}
        disabled={disabled}
        type="checkbox"
        name={group}
        onChange={onChange}
        defaultChecked={defaultChecked}
        {...otherProps}
      />
      <ToggleElement>
        <span>{children}</span>
      </ToggleElement>
    </CheckWrapper>
  )
}

const CheckWrapper = styled.label`
  min-height: ${targetSize.normal};
  padding-top: ${spacing.tiny};
  padding-bottom: ${spacing.tiny};
  display: block;
`

const ToggleElement = styled.div`
  background: ${color.line};
  border-radius: 2em;
  padding: 2px;
  transition: all 0.4s ease;
  span {
    position: absolute;
    clip: rect(0, 0, 0, 0);
    clip: rect(0 0 0 0);
  }
  &:after {
    border-radius: 50%;
    background: ${color.white};
    transition: all 0.2s ease;
  }
`

const Check = styled.input`
  position: absolute;
  clip: rect(0, 0, 0, 0);
  clip: rect(0 0 0 0);

  &,
  &:after,
  &:before,
  & *,
  & *:after,
  & *:before,
  & + ${ToggleElement} {
    box-sizing: border-box;
    &::selection {
      background: none;
    }
  }

  + ${ToggleElement} {
    outline: 0;
    display: block;
    width: 4em;
    height: 2em;
    position: relative;
    cursor: pointer;
    user-select: none;
    &:after,
    &:before {
      position: relative;
      display: block;
      content: "";
      width: 50%;
      height: 100%;
    }

    &:after {
      left: 0;
    }

    &:before {
      display: none;
    }
  }

  &:checked + ${ToggleElement} {
    background: ${color.green};
    &:after {
      transform: translateX(100%);
    }
  }
`

Toggle.propTypes = {
  disabled: PropTypes.bool,
  group: PropTypes.string,
  id: PropTypes.string.isRequired,
  input: PropTypes.any,
  onChange: PropTypes.func,
  className: PropTypes.string,
  defaultChecked: PropTypes.bool,
  children: PropTypes.any
}

Toggle.defaultProps = {
  disabled: false,
  input: {},
  onChange: null,
  value: "",
  className: "",
  defaultChecked: false,
  children: ""
}

export default Toggle
