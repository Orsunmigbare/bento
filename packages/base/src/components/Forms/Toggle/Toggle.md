```jsx
<div>
  <Label htmlFor="gfdfgfddffhhg">Toggle This</Label>
  <Toggle group="testddd" id="gfdfgfddffhhg" onChange={() => null}>
    Toggle this
  </Toggle>
</div>
```
