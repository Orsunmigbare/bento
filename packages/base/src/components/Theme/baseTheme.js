export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
