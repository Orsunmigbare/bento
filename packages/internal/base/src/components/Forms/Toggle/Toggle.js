import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  targetSize,
  spacing,
  borderRadius,
  fontFamily,
  font,
  color,
  themify
} from "@staccx/theme"

const Toggle = ({
  children,
  id,
  group,
  disabled,
  onChange,
  className,
  defaultChecked,
  ...otherProps
}) => {
  return (
    <ToggleWrapper htmlFor={id} className={className}>
      <ToggleInput
        id={id}
        disabled={disabled}
        type="checkbox"
        name={group}
        onChange={onChange}
        defaultChecked={defaultChecked}
        {...otherProps}
      />
      <ToggleElement>
        <span>{children}</span>
      </ToggleElement>
    </ToggleWrapper>
  )
}

export const TOGGLE = "TOGGLE"
const ToggleWrapper = styled.label`
  min-height: ${targetSize.normal};
  padding-top: ${spacing.tiny()};
  padding-bottom: ${spacing.tiny()};
  display: block;
  ${themify(TOGGLE)};
`

export const TOGGLE_ELEMENT = "TOGGLE_ELEMENT"
const ToggleElement = styled.div`
  background: ${color.line};
  border-radius: 2em;
  padding: 2px;
  transition: all 0.4s ease;
  span {
    position: absolute;
    clip: rect(0, 0, 0, 0);
    clip: rect(0 0 0 0);
  }
  &:after {
    border-radius: 50%;
    background: ${color.white};
    transition: all 0.2s ease;
  }
  ${themify(TOGGLE_ELEMENT)};
`

export const TOGGLE_INPUT = "TOGGLE_INPUT"
export const TOGGLE_ELEMENT_CHECKED = "TOGGLE_ELEMENT_CHECKED"
const ToggleInput = styled.input`
  position: absolute;
  clip: rect(0, 0, 0, 0);
  clip: rect(0 0 0 0);

  &,
  &:after,
  &:before,
  & *,
  & *:after,
  & *:before,
  & + ${ToggleElement} {
    box-sizing: border-box;
    &::selection {
      background: none;
    }
  }

  + ${ToggleElement} {
     {
      /* TODO: Burde dette v√¶rt flyttet inn i Toggle-element over?*/
    }
    outline: 0;
    display: block;
    width: 4em;
    height: 2em;
    position: relative;
    cursor: pointer;
    user-select: none;
    &:after,
    &:before {
      position: relative;
      display: block;
      content: "";
      width: 50%;
      height: 100%;
    }

    &:after {
      left: 0;
    }

    &:before {
      display: none;
    }
  }

  &:checked + ${ToggleElement} {
    background: ${color.green};
    &:after {
      transform: translateX(100%);
    }
    ${themify(TOGGLE_ELEMENT_CHECKED)};
  }
  ${themify(TOGGLE_INPUT)};
`

Toggle.propTypes = {
  disabled: PropTypes.bool,
  group: PropTypes.string,
  id: PropTypes.string.isRequired,
  input: PropTypes.any,
  onChange: PropTypes.func,
  className: PropTypes.string,
  defaultChecked: PropTypes.bool,
  children: PropTypes.any
}

Toggle.defaultProps = {
  disabled: false,
  input: {},
  onChange: null,
  value: "",
  className: "",
  defaultChecked: false,
  children: ""
}

export default Toggle
