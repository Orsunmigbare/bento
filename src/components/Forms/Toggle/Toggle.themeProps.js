import { themePropTypes } from "../../../theming"

const themeProps = {
  wrapper: {
    name: "TOGGLE",
    description: "Toggle style",
    type: themePropTypes.style
  },
  toggleElement: {
    name: "TOGGLE_ELEMENT",
    description: "Toggle element style",
    type: themePropTypes.style
  },
  input: {
    name: "TOGGLE_INPUT",
    description: "Toggle input style",
    type: themePropTypes.style
  },
  checked: {
    name: "  TOGGLE_ELEMENT_CHECKED",
    description: "Toggle element checked style",
    type: themePropTypes.style
  }
}

export default themeProps
