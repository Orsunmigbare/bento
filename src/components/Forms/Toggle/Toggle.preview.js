import React from "react"
import Toggle from "./Toggle"
import Label from "../Label/Label"

const preview = {
  title: "Toggle",
  category: "Components/Forms",
  component: Toggle,
  render: props => (
    <div>
      <Label htmlFor="gfdfgfddffhhg">Toggle This</Label>
      <Toggle group="testddd" id="gfdfgfddffhhg" onChange={() => null}>
        Toggle this
      </Toggle>
    </div>
  )
}

export default preview
