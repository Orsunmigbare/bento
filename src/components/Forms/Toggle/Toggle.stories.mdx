import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Toggle from "./Toggle"


<Meta title="Components|Forms/Toggle" component={Toggle}/>

# Toggle
Toggle is a component often used to activate and deactivate something. 

```javascript
import { Toggle } from "@staccx/bento"
```

## Standard Toggle

<Preview>
  <Story name="Standard">
            <Toggle name="Toggle"/>
  </Story>
</Preview>

## Toggle group

<Preview>
  <Story name="Group">
  <>
            <Toggle id="1" group="test"/>
            <Toggle id="2" group="test"/>
            <Toggle id="3" group="test"/>
    </>
  </Story>
</Preview>



## Props for Toggle

<Props of={Toggle} />
