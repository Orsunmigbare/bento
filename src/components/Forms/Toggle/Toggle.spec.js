import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Toggle from "./Toggle"

describe("Toggle", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("disabled", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle disabled />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("group", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle group="Group" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("id", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle id="Group" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("input", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle input="" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("input", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle onChange={e => console.log(e.target.value)} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Toggle", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle className="Toggle" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("defaultChecked", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle defaultChecked />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle>
              <p>hei</p>
            </Toggle>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("All", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Toggle
              disabled
              group="Group"
              input=""
              onChange={e => console.log(e.target.value)}
              className="Toggle"
              defaultChecked
            />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
