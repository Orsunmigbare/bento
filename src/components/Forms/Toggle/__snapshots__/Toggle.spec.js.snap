// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshots All 1`] = `
<label
  className="sc-AxjAm juJlSJ Toggle"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={true}
    disabled={true}
    name="Group"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots Children 1`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      <p>
        hei
      </p>
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots No props 1`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots Toggle 1`] = `
<label
  className="sc-AxjAm juJlSJ Toggle"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots defaultChecked 1`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={true}
    disabled={false}
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots disabled 1`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={true}
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots group 1`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    name="Group"
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots id 1`] = `
<label
  className="sc-AxjAm juJlSJ"
  htmlFor="Group"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    id="Group"
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots input 1`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    onChange={null}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;

exports[`Toggle Snapshots input 2`] = `
<label
  className="sc-AxjAm juJlSJ"
>
  <input
    className="sc-AxiKw jvsQrJ"
    defaultChecked={false}
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="sc-AxirZ iudxgh"
  >
    <span>
      
    </span>
  </div>
</label>
`;
